src/main/java/com/code_intelligence/jazzer/mutation/mutator/lang/IntegralMutatorFactory.java

```diff
@@ -289,8 +289,26 @@ protected final long mutateImpl(long value, PseudoRandom prng) {
       return value;
     }
 
-    protected final long crossOverImpl(long value, long otherValue, PseudoRandom prng) {
-      return value;
+    protected final long crossOverImpl(long x, long y, PseudoRandom prng) {
+      switch (prng.indexIn(4)) {
+        case 0:
+          return mean(x, y);
+        case 1:
+          return forceInRange(x | y);
+        case 2:
+          return forceInRange(x & y);
+        case 3:
+          return forceInRange(x ^ y);
+        default:
+          throw new AssertionError("Invalid cross over function.");
+      }
+    }
+
+    private static long mean(long x, long y) {
```

Comment:
Shouldn't the following implementation suffice?

```java
private static long mean(long x, long y) {
    return (x & y) + ((x ^ y) >> 1);
}
```

With that, when we add the common set bits (`x & y`) and the half of the sum of the differing bits together (`(x ^ y) >> 1`), the result will never exceed the sum of `x` and `y` as both parts of the calculation are guaranteed to be smaller than or equal to `x` and `y`.

Reply:
Rounding negative values does "not quite work", it's discussed in this SO post: https://stackoverflow.com/questions/16115501/mean-of-two-ints-or-longs-without-overflow-truncating-towards-0

I don't really prefer any solution as rounding shouldn't be important in our use-case. Do you think the other one is faster and should be preferred?

Reply:
Ah I didn't think of that.. Then we should keep the original implementation. Performance wise, I don't think there's much of an issue. the local variables are hopefully optimized accordingly, so the remaining operations are ones that are quick to execute :)